cmd/sysl/cmd_codegen.go

```diff
@@ -70,3 +84,25 @@ func (p *codegenCmd) Execute(args ExecuteArgs) error {
 	}
 	return outputToFiles(output, syslutil.NewChrootFs(args.Filesystem, p.outDir))
 }
+
+func (p *codegenCmd) loadFlags() error {
+	err := validate.CodeggenRequiredFlags(p.config, p.grammar, p.transform)
```

Comment:
1) typo in CodegenRequiredFlags
2) Why did you choose too put that function in the validate package? It is only used by this command, and is rightly coupled to this command, it really shuold just be here.

Reply:
This method is added to check 

> As mentioned, I don't see the value-proposition of this change. You're simply moving command arguments from the command line to a seperate file which does very little as far as user experience is concerned.
> 
> Now if thats all we want to do then I have a much better solution. gcc (and other programs) support a `@file` command line option which just reads the contents of the file as the command line arguments (it is used because gcc can get REALLY long command lines with dozens of flags). How about simply supporting that and using it for all the sysl commands as an option?
> 
> see the very bottom of https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html#Overall-Options ( @file)
> 
> how you'd d this:
> in `cmd/sysl/sysl.go main3()` you'd scan the supplied args list, for any arg which looks like `@filename` , the filename would be looked up, if it exists it gets read into the args in the current position replacing the @filename flag, _then_ once they are all consumed pass the new command line args to kingpin for processing. none of the actual commands need to be changed to support this.

@anz-gordonj7 , this change is a simple fixing to make sure it can be delivered soon and @joshcarp can use it in next week's show case. And after this is done, I will change more on this piece, I think then I can try your solution. Thoughts?

Reply:
validate package is setup from validation purpose, and method CodeggenRequiredFlags is used to do validation too, so I add it here to reduce code in cmd_codgen.go.

Reply:
> @anz-gordonj7 , this change is a simple fixing to make sure it can be delivered soon and @joshcarp can use it in next week's show case. And after this is done, I will change more on this piece, I think then I can try your solution. Thoughts?

I honestly never want to hear the phrase "is a simple fixing to make sure it can be delivered soon". nothing is stopping @joshcarp from using your branch in a demo (or some other hack/fix), doesnt mean it needs to be in master. I don't want to see half-patches merged thank you.

> validate package is setup from validation purpose, and method CodeggenRequiredFlags is used to do validation too

It is *supposed* to be validating a sysl file, not the command line options. This is absolutly the wrong place to put this code.

Reply:
@anz-gordonj7 , thank you for your update and you are right. But simple fixing doesn't mean fixing with bad implementation or code, right? I raise PR because I am considering team will use this feature after @joshcarp's showcase. And anyway I have pinged @joshcarp and see if he can use my branch to do showcase.

Reply:
@anz-gordonj7, and in the same time,  I am trying your suggestion :
how you'd d this:
in cmd/sysl/sysl.go main3() you'd scan the supplied args list, for any arg which looks like @filename , the filename would be looked up, if it exists it gets read into the args in the current position replacing the @filename flag, then once they are all consumed pass the new command line args to kingpin for processing. none of the actual commands need to be changed to support this.

Reply:
Fixed in another branch issue-619-new-fixing